crates/autopilot/src/domain/settlement/mod.rs

```diff
@@ -76,12 +112,59 @@ impl Settlement {
 
 #[derive(Debug, thiserror::Error)]
 pub enum Error {
+    #[error("failed communication with the database: {0}")]
+    DatabaseError(sqlx::Error),
     #[error(transparent)]
     Solution(#[from] solution::Error),
     #[error("settlement refers to an auction from a different environment")]
     WrongEnvironment,
-    #[error("connection to the persistence layer failed: {0}")]
-    PersistenceConnection(#[from] infra::persistence::Error),
+    #[error("auction not found in the database")]
+    MissingAuction,
+    #[error("failed to get fee policy from database: {0} for order: {1}")]
+    FeePolicy(infra::persistence::dto::fee_policy::Error, domain::OrderUid),
+    #[error("failed to get price from database for token: {0}")]
+    Price(eth::TokenAddress),
+    #[error("winner mismatch: expected competition winner {expected}, settlement solver {got}")]
+    WinnerMismatch {
+        expected: eth::Address,
+        got: eth::Address,
+    },
+    #[error("failed to get score from database for a coresponding competition solution, err: {0}")]
+    InvalidScore(anyhow::Error),
     #[error(transparent)]
-    Auction(#[from] infra::persistence::error::Auction),
+    Score(#[from] solution::error::Score),
+    #[error("failed to get competition data from database {0}")]
+    SolverCompetition(anyhow::Error),
```

Comment:
So this is a Database error as well, isn't it?

Reply:
No, this should cover errors related to conversion of solver competition data to solution. refactor so let's see if better now.